### PR TITLE
Issue #115: Update DeveloperGuide with `Find` command implementation

### LINKED ISSUE #115: Update DG on implementation for `find` command
Week 10's task - DG individual draft.

@charmainehly char let's KIV the `sort` command.

### PR DESCRIPTION
**Pull request by charmainehly**
Currently, the DeveloperGuide does not contain details on the implementation of the find feature.

There is no documentation on important design and logic considerations.

Let's note down important details for the find feature
on the following: (work in progress)
* what the feature is about
* implementation of the feature
* why is the implementation of the feature as such
* relevant UML diagrams

**Note to team:** Perhaps we can consider standardising the format of this particular section! I will still be updating the documentation for this feature as we go!

Closes #115

### CODE REVIEW

**Review by LeeEnEn — APPROVED** [review]
Format is very clean and neat! I will definitely use this for future references! 🤣

**Review by lzan98 — APPROVED** [review]
Neat! Will merge it in now.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,60 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Find feature
+
+#### What is the feature about?
+The `find` mechanism is facilitated by `AddressBook3`. The `find` command in `TAlent Assistant™`
```
Comment (domlimm) [inline docs/DeveloperGuide.md:160]: Hey @charmainehly, I think this should be `AddressBook`. If I'm not mistaken, it's referencing to the `AddressBook` class.

Comment (charmainehly) [reply docs/DeveloperGuide.md:160]: Got it! Let me fix this in a future commit and PR. Thanks for catching this :)

**Pull request merged**